### ISSUE TITLE
test cases failed in Java 11 and Java 16 on 2 computers

### ISSUE THREAD

**Issue by colourful-land** [TokenScript/attestation#159]
On my PC:
````
vnaeli@vnaeli-ga990fxaud5:~/IdeaProjects/attestation$ java -version
openjdk version "11.0.11" 2021-04-20
OpenJDK Runtime Environment (build 11.0.11+9-Ubuntu-0ubuntu2.20.04)
OpenJDK 64-Bit Server VM (build 11.0.11+9-Ubuntu-0ubuntu2.20.04, mixed mode, sharing)
vnaeli@vnaeli-ga990fxaud5:~/IdeaProjects/attestation$ javac -version
javac 11.0.11
vnaeli@vnaeli-ga990fxaud5:~/IdeaProjects/attestation$ gradle build

> Task :test

TestAttestationUsageEip712 > otherTimeLimit() FAILED
    org.opentest4j.AssertionFailedError at TestAttestationUsageEip712.java:103

TestAttestationUsageEip712 > testDecoding() FAILED
    org.opentest4j.AssertionFailedError at TestAttestationUsageEip712.java:112

228 tests completed, 2 failed

> Task :test FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':test'.
> There were failing tests. See the report at: file:///home/vnaeli/IdeaProjects/attestation/build/reports/tests/test/index.html

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 26s
4 actionable tasks: 1 executed, 3 up-to-date
````

**Comment by colourful-land** [comment]
It seems one of the previous fix I applied actually reduced the failed test-cases. An earlier version had more:

````
attestation$ gradle build

> Task :compileJava
Note: /home/vnaeli/IdeaProjects/attestation/src/main/java/com/alphawallet/attestation/IdentifierAttestation.java uses or overrides a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
Note: Some input files use unchecked or unsafe operations.
Note: Recompile with -Xlint:unchecked for details.

> Task :compileTestJava
Note: Some input files use unchecked or unsafe operations.
Note: Recompile with -Xlint:unchecked for details.

> Task :test

TestAttestationUsageEip712 > otherTimeLimit() FAILED
    org.opentest4j.AssertionFailedError at TestAttestationUsageEip712.java:103

Eip712Test > referenceEncodingOtherOrder() FAILED
    org.opentest4j.AssertionFailedError at Eip712Test.java:58

Eip712Test > referenceEncoding() FAILED
    org.opentest4j.AssertionFailedError at Eip712Test.java:73

228 tests completed, 3 failed

> Task :test FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':test'.
> There were failing tests. See the report at: file:///home/vnaeli/IdeaProjects/attestation/build/reports/tests/test/index.html

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 25s
4 actionable tasks: 4 executed
````

**Comment by colourful-land** [comment]
The same issue is reproduced if I use java/javac from jdk16
````
vnaeli@vnaeli-ga990fxaud5:~/IdeaProjects/attestation$ javac -version
javac 16.0.1
vnaeli@vnaeli-ga990fxaud5:~/IdeaProjects/attestation$ java -version
openjdk version "16.0.1" 2021-04-20
OpenJDK Runtime Environment (build 16.0.1+9-Ubuntu-120.04)
OpenJDK 64-Bit Server VM (build 16.0.1+9-Ubuntu-120.04, mixed mode, sharing)
vnaeli@vnaeli-ga990fxaud5:~/IdeaProjects/attestation$ gradle build

> Task :test

Eip712Test > referenceEncodingOtherOrder() FAILED
    org.opentest4j.AssertionFailedError at Eip712Test.java:58

Eip712Test > referenceEncoding() FAILED
    org.opentest4j.AssertionFailedError at Eip712Test.java:73

228 tests completed, 2 failed

> Task :test FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':test'.
> There were failing tests. See the report at: file:///home/vnaeli/IdeaProjects/attestation/build/reports/tests/test/index.html

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 23s
4 actionable tasks: 1 executed, 3 up-to-date
````

**Comment by samuelhbne** [comment]
Could not reproduce this on Ubuntu-20.04 AMD64 with Java-11.0.11 and Gradle 7.1.1

```
sam@focal64vm0:~$ java -version
openjdk version "11.0.11" 2021-04-20
OpenJDK Runtime Environment (build 11.0.11+9-Ubuntu-0ubuntu2.20.04)
OpenJDK 64-Bit Server VM (build 11.0.11+9-Ubuntu-0ubuntu2.20.04, mixed mode, sharing)
sam@focal64vm0:~$ gradle -version

------------------------------------------------------------
Gradle 7.1.1
------------------------------------------------------------

Build time:   2021-07-02 12:16:43 UTC
Revision:     774525a055494e0ece39f522ac7ad17498ce032c

Kotlin:       1.4.31
Groovy:       3.0.7
Ant:          Apache Ant(TM) version 1.10.9 compiled on September 27 2020
JVM:          11.0.11 (Ubuntu 11.0.11+9-Ubuntu-0ubuntu2.20.04)
OS:           Linux 5.4.0-77-generic amd64

sam@focal64vm0:~$ git clone https://github.com/TokenScript/attestation.git
Cloning into 'attestation'...
remote: Enumerating objects: 8264, done.
remote: Counting objects: 100% (653/653), done.
remote: Compressing objects: 100% (454/454), done.
remote: Total 8264 (delta 221), reused 420 (delta 118), pack-reused 7611
Receiving objects: 100% (8264/8264), 10.63 MiB | 1.80 MiB/s, done.
Resolving deltas: 100% (3523/3523), done.
sam@focal64vm0:~$ cd attestation/
sam@focal64vm0:~/attestation$ gradle build
Starting a Gradle Daemon (subsequent builds will be faster)

> Task :compileJava
Note: /home/sam/attestation/src/main/java/com/alphawallet/attestation/IdentifierAttestation.java uses or overrides a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
Note: Some input files use unchecked or unsafe operations.
Note: Recompile with -Xlint:unchecked for details.

> Task :compileTestJava
Note: Some input files use unchecked or unsafe operations.
Note: Recompile with -Xlint:unchecked for details.

BUILD SUCCESSFUL in 1m 59s
5 actionable tasks: 5 executed
sam@focal64vm0:~/attestation$ 
```

**Comment by samuelhbne** [comment]
Could not reproduce this on Ubuntu-20.04 AMD64 with Java-16.0.1 and Gradle 7.1.1

```
sam@focal64vm0:~$ java -version
openjdk version "16.0.1" 2021-04-20
OpenJDK Runtime Environment (build 16.0.1+9-Ubuntu-120.04)
OpenJDK 64-Bit Server VM (build 16.0.1+9-Ubuntu-120.04, mixed mode, sharing)
sam@focal64vm0:~$ gradle -version

------------------------------------------------------------
Gradle 7.1.1
------------------------------------------------------------

Build time:   2021-07-02 12:16:43 UTC
Revision:     774525a055494e0ece39f522ac7ad17498ce032c

Kotlin:       1.4.31
Groovy:       3.0.7
Ant:          Apache Ant(TM) version 1.10.9 compiled on September 27 2020
JVM:          16.0.1 (Private Build 16.0.1+9-Ubuntu-120.04)
OS:           Linux 5.4.0-77-generic amd64

sam@focal64vm0:~$ git clone https://github.com/TokenScript/attestation.git
Cloning into 'attestation'...
remote: Enumerating objects: 8264, done.
remote: Counting objects: 100% (653/653), done.
remote: Compressing objects: 100% (454/454), done.
remote: Total 8264 (delta 221), reused 420 (delta 118), pack-reused 7611
Receiving objects: 100% (8264/8264), 10.63 MiB | 1.46 MiB/s, done.
Resolving deltas: 100% (3523/3523), done.
sam@focal64vm0:~$ cd attestation/
sam@focal64vm0:~/attestation$ gradle build

> Task :compileJava
Note: Some input files use or override a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
Note: Some input files use unchecked or unsafe operations.
Note: Recompile with -Xlint:unchecked for details.

> Task :compileTestJava
Note: Some input files use unchecked or unsafe operations.
Note: Recompile with -Xlint:unchecked for details.

BUILD SUCCESSFUL in 25s
5 actionable tasks: 5 executed
sam@focal64vm0:~/attestation$ 
```

**Comment by colourful-land** [comment]
could reproduce it with Ubuntu-20.04 AMD64 with Java-16.0.1 and Gradle 7.1.1

**Comment by oleggrib** [comment]
I have got issue with tests for Windows 10

>gradle -v

------------------------------------------------------------
Gradle 6.7.1
------------------------------------------------------------

Build time:   2020-11-16 17:09:24 UTC
Revision:     2972ff02f3210d2ceed2f1ea880f026acfbab5c0

Kotlin:       1.3.72
Groovy:       2.5.12
Ant:          Apache Ant(TM) version 1.10.8 compiled on May 10 2020
JVM:          15.0.1 (Oracle Corporation 15.0.1+9-18)
OS:           Windows 10 10.0 amd64


>gradle testJavaScript

> Task :npmInstall
npm WARN read-shrinkwrap This version of npm is compatible with lockfileVersion@1, but package-lock.json was generated for lockfileVersion@2. I'll try to do my best with it!
npm WARN tokenscript-attestation@0.0.1 No repository field.
npm WARN tokenscript-attestation@0.0.1 No license field.

audited 6 packages in 0.933s
found 0 vulnerabilities


> Task :test

TestAttestationUsageEip712 > testSunshine() FAILED
    org.opentest4j.AssertionFailedError at TestAttestationUsageEip712.java:93

TestAttestationUsageEip712 > nonverifiableUseAttesation() FAILED
    org.opentest4j.AssertionFailedError at TestAttestationUsageEip712.java:219

TestAttestationUsageEip712 > otherTimeLimit() FAILED
    org.opentest4j.AssertionFailedError at TestAttestationUsageEip712.java:103

TestAttestationUsageEip712 > testDecoding() FAILED
    org.opentest4j.AssertionFailedError at TestAttestationUsageEip712.java:112

IdentifierAttestationTest > testUrlEncoding() FAILED
    org.opentest4j.AssertionFailedError at IdentifierAttestationTest.java:160

228 tests completed, 5 failed

> Task :test FAILED

**Comment by colourful-land** [comment]
I once fixed one of the 4 errors I had (that's why I only have 3 errors). The fix I had is:

https://github.com/TokenScript/attestation/commit/52c6b261751e85b407fa0aefc31705f9593e1526

I hope this might shed some light on the cause of these problems.

**Comment by colourful-land** [comment]
fixed by #164
